step-release-vis/src/app/services/environment.ts

```diff
@@ -1,19 +1,26 @@
 import {Injectable} from '@angular/core';
 import {Polygon} from '../models/Polygon';
 import {CandidateInfo, Environment} from '../models/Data';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {FileService} from './file';
 
 // TODO(naoai): Iterate and calculate the polygons
 @Injectable({
   providedIn: 'root'
 })
 export class EnvironmentService {
 
-  constructor() {
+  constructor(private fileService: FileService) {
   }
 
   // xs: 0-100, ys: timestamps
-  getPolygons(jsonFile: string): Polygon[] {
-    const environments = this.readJson(jsonFile);
+  getPolygons(jsonFile: string): Observable<Polygon[]> {
```

Comment:
test for this?

Reply:
Testing Observables requires research. I thought I'd add it later not to block other people.

Reply:
// TODO(...) ... :)

Reply:
My bad, added the TODO.